### PR TITLE
Update contribution guidelines.

### PR DESCRIPTION
**Pull request by geeknoid**
Minor update to our contribution guidelines.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -48,7 +48,10 @@ permission to use and redistribute your contributions as part of the project.
 Any substantial design deserves a design document. Design documents are written with Google Docs and
 should be shared with the community by adding the doc to our [Team Drive](https://drive.google.com/corp/drive/u/0/folders/0AIS5p3eW9BCtUk9PVA)
 and sending a note to the appropriate working group to let people know the doc is there. To get write access
-to the drive, you'll need to be an official contributor to the Istio project per GitHub.
+to the drive, you'll need to be a [member](ROLES.md#member) of the Istio organization.
+
+We have a common [template](https://docs.google.com/document/d/1cpolPNH_RtSobUjTkXyyxsCJT7EKD8fGiN2TgYLeJu8/edit#heading=h.7zgnj8bwqfld).
```
Comment (smawson) [inline CONTRIBUTING.md]: We have a common design doc template.

Maybe [design doc template] as the link.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -59,14 +62,29 @@ but there is a human involved there, so please forgive any delays.
...
-the GitHub issue.
-- Once there is general agreement on the technical direction, submit a PR.
-- Contribute documentation of the feature to the [istio.io](https://istio.io) repo (https://github.com/istio/istio.github.io).
+about the requirements and use cases that it is trying to address. Include a discussion of the proposed design and technical details of the
+implementation in the issue.
+
+- If the feature is substantial enough:
+
+  - Working group leads will ask for a design document as outlined in the previous section.
+  Create this document and add a link to it in the GitHub issue. Don't forget to send a note to the
```
Comment (smawson) [inline CONTRIBUTING.md]: s/this document/the design document/

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -59,14 +62,29 @@ but there is a human involved there, so please forgive any delays.
...
+  Create this document and add a link to it in the GitHub issue. Don't forget to send a note to the 
+  working group to let everyone know your document is ready for review.
+
+  - Depending of the breath of the design and how contentious it is, the working group leads may decide
+  the feature needs to be discussed in one or more working group meetings before being approved.
+
+  - Once the major technical issues are resolved and agreed upon, post a note to the working group's mailing
+  list with the design decision and the general execution plan.
+
+- Submit a PR to istio/istio with your code changes.
```
Comment (smawson) [inline CONTRIBUTING.md]: one or more PRs?  Something about breaking into manageable chunks? Or too much detail here?

Could just say:

Submit PRs to istio/istio with your code changes.

Maybe same with documentation (s/a PR/PRs/)

**Comment by geeknoid** [comment]
Addressed Sven't feedback.

File: REVIEWING.md [hunk 4]
```diff
@@ -78,20 +78,13 @@ review, withdraws their hold, or is overruled by a preponderance of maintainers.
...
-to the project. Active contributors to the project will eventually get noticed
-by the maintainers, and one of the existing maintainers will nominate the
-contributor to become a maintainer.  A 'yes' vote from >75% of
-the existing maintainers is required for approval.
-
-Removing a maintainer requires a 'yes' vote from >75% of the exsiting
-maintainers. Note that inactive maintainers might periodically be removed
-simply to keep the membership list accurate.
+to the project. Please see our [community roles](https://github.com/istio/community/blob/master/ROLES.md)
+document for information on how this is done.
```
Comment (linsun) [inline REVIEWING.md]: just realized that you don't have maintainer role in the community roles url.   you may want to change it to the mapping role.

Comment (geeknoid) [reply REVIEWING.md]: I'd like to wait on updating this doc until the current experiment with the CODEOWNERS stuff is done. If we fully switch to CODEOWNERS, then ROLES.md needs to be updated to merge the reviewer & approver roles, and the wording here will need to change too.

Comment (linsun) [reply REVIEWING.md]: SGTM.

**Pull request merged**